api/src/pcapi/core/bookings/api.py

```diff
@@ -460,15 +462,29 @@ def cancel_booking_by_offerer(booking: Booking) -> None:
 
 
 def cancel_bookings_from_stock_by_offerer(stock: offers_models.Stock) -> list[Booking]:
-    cancelled_bookings = _cancel_bookings_from_stock(stock, BookingCancellationReasons.OFFERER)
+    cancelled_bookings = _cancel_bookings_from_stock(
+        stock, BookingCancellationReasons.OFFERER, typing.cast(bool, stock.offer.isEvent)
+    )
+    search.async_index_offer_ids([stock.offerId])
+    return cancelled_bookings
+
+
+def cancel_bookings_from_stock_no_longer_free(stock: offers_models.Stock) -> list[Booking]:
+    cancelled_bookings = _cancel_bookings_from_stock(
+        stock,
+        BookingCancellationReasons.OFFERER,
+        stock.offer.subcategoryId in constants.FREE_OFFER_SUBCATEGORY_IDS_TO_ARCHIVE,
+    )
     search.async_index_offer_ids([stock.offerId])
     return cancelled_bookings
 
 
 def cancel_bookings_from_rejected_offer(offer: offers_models.Offer) -> list[Booking]:
     cancelled_bookings = []
     for stock in offer.stocks:
-        cancelled_bookings.extend(_cancel_bookings_from_stock(stock, BookingCancellationReasons.FRAUD))
+        cancelled_bookings.extend(
+            _cancel_bookings_from_stock(stock, BookingCancellationReasons.FRAUD, typing.cast(bool, stock.offer.isEvent))
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
**💣 chore:** ‏Pas besoin du cast non plus